### PR TITLE
[WIP] Subscription Edge Case

### PR DESCRIPTION
**Pull request by thefringeninja**
Added some additional tests to ensure a stream / all can be subscribed to before anything has been written.

Also found a bug concerning static usage of SHA1.

### CODE REVIEW

**Comment by thefringeninja** [comment]
For some reason `Can_subscribe_to_a_stream_from_start_before_events_are_written` is failing.

File: src/Cedar.EventStore.MsSql2008/MsSqlEventStore.cs [hunk 1]
```diff
@@ -193,7 +193,6 @@ protected override void Dispose(bool disposing)
 
         private struct StreamIdInfo
         {
-            private static readonly SHA1 s_sha1 = SHA1.Create();
```
Comment (damianh) [inline src/Cedar.EventStore.MsSql2008/MsSqlEventStore.cs]: Nice catch.

**Comment by damianh** [comment]
Thx for the PR. Is this a WIP?

**Comment by damianh** [comment]
The SHA1 fix should be a separate PR, it's not related to the subscription issue.

**Comment by damianh** [comment]
I've cherry picked the SHA1 fix

**Comment by damianh** [comment]
Cherry picked the sample failing test and tweaked it a bit. I've fixed the issue and should be in next CI build. Cheers.
